zipline/assets/assets.py

```diff
@@ -1026,6 +1030,28 @@ def get_ordered_contracts(self, root_symbol):
             self._ordered_contracts[root_symbol] = oc
             return oc
 
+    def offset_of_contract(self, asset, date):
+        """
+        Get the offset of the given contract relative to the contract with an
+        offset of zero on the given date.
+        """
+        oc = self.get_ordered_contracts(asset.root_symbol)
+        current_contract_sid = oc.contract_before_auto_close(date.value)
```

Comment:
The `.__sub__` is an interesting thought! I can definitely see that being useful in making this code more concise, my only worry is that it might be just a bit _too_ clever. That is, having not seen it before it's slightly more difficult to reason about than the verbose, explicit code. Regardless, in heeding your final comment, I think this would be getting a tad too out of scope for this PR.